engine/runtime-instrument-id-execution/src/main/java/org/enso/interpreter/instrument/IdExecutionInstrument.java

```diff
@@ -250,6 +250,7 @@ public void onReturnExceptional(VirtualFrame frame, Throwable exception) {
       }
     }
 
+    @CompilerDirectives.TruffleBoundary
```

Comment:
Shouldn't this be on `setExecuted` instead?

Reply:
Right